lib/src/front_end/ast_node_visitor.dart

```diff
@@ -618,7 +619,38 @@ class AstNodeVisitor extends ThrowingAstVisitor<void>
 
   @override
   void visitInstanceCreationExpression(InstanceCreationExpression node) {
-    throw UnimplementedError();
+    token(node.keyword, after: space);
+
+    // If there is an import prefix and/or constructor name, then allow
+    // splitting before the `.`. This doesn't look good, but is consistent with
+    // constructor calls that don't have `new` or `const`. We allow splitting
+    // in the latter because there is no way to distinguish syntactically
+    // between a named constructor call and any other kind of method call.
+    var calls = <Piece>[];
+
+    if (node.constructorName.type.importPrefix case var importPrefix?) {
+      token(importPrefix.name);
+      calls.add(pieces.split());
+      token(importPrefix.period);
+    }
+
+    token(node.constructorName.type.name2);
```

Comment:
Nit: Maybe make a local for `node.constructorName.type` to make it easier to read this section?

Reply:
Good call. Did the same for `node.constructorName` too.